src/components/Griff/index.tsx

```diff
@@ -0,0 +1,444 @@
+      yAccessor,
+      y0Accessor,
+      y1Accessor,
+      timeDomain,
+      timeSubDomain,
+      xDomain,
+      xSubDomain,
+      yDomain,
+      ySubDomain,
+      pointWidth,
+      strokeWidth,
+      opacity,
+      opacityAccessor,
+      pointWidthAccessor,
+    } = this.props;
+    const { collectionsById } = this.state;
+    return Object.keys(collectionsById).reduce((acc: BaseCollection[], id) => {
+      if (!memberCount[id]) {
+        return acc;
+      }
+      const collection = collectionsById[id];
+      const dataProvider = {
+        drawLines,
+        drawPoints,
+        pointWidth,
+        strokeWidth,
+        opacity,
+        opacityAccessor,
+        pointWidthAccessor,
+        timeAccessor,
+        xAccessor,
+        x0Accessor,
+        x1Accessor,
+        yAccessor,
+        y0Accessor,
+        y1Accessor,
+        timeDomain,
+        timeSubDomain,
+        xDomain,
+        xSubDomain,
+        yDomain,
+        ySubDomain,
+      };
+      const completedCollection: BaseCollection = combineItems(
+        DEFAULT_SERIES_CONFIG,
+        dataProvider,
+        collection
+      );
+      return [...acc, completedCollection];
+    }, []);
+  };
+
+  getSeriesObjects = () => {
+    const {
+      drawLines,
+      drawPoints,
+      loader,
+      timeAccessor,
+      xAccessor,
+      x0Accessor,
+      x1Accessor,
+      yAccessor,
+      y0Accessor,
+      y1Accessor,
+      xDomain,
+      xSubDomain,
+      yDomain,
+      ySubDomain,
+      pointWidth,
+      strokeWidth,
+      opacity,
+      opacityAccessor,
+      pointWidthAccessor,
+      timeDomain,
+      timeSubDomain,
+      updateInterval,
+    } = this.props;
+    const { collectionsById, seriesById } = this.state;
+    return Object.keys(seriesById).reduce((acc: BaseSeries[], id) => {
+      const series = seriesById[id];
+      const dataProvider: IncomingItem = {
+        id,
+        drawLines,
+        drawPoints,
+        loader,
+        pointWidth,
+        strokeWidth,
+        opacity,
+        opacityAccessor,
+        pointWidthAccessor,
+        timeAccessor,
+        xAccessor,
+        x0Accessor,
+        x1Accessor,
+        yAccessor,
+        y0Accessor,
+        y1Accessor,
+        timeDomain,
+        timeSubDomain,
+        xDomain,
+        xSubDomain,
+        yDomain,
+        ySubDomain,
+        updateInterval,
+      };
+      const collection: IncomingCollection =
+        series.collectionId !== undefined
+          ? collectionsById[series.collectionId] || { id: '' }
+          : { id: '' };
+      const completedSeries: BaseSeries = combineItems(
+        DEFAULT_SERIES_CONFIG,
+        dataProvider,
+        collection,
+        series
+      );
+      completedSeries.timeDomain = copyDomain(
+        completedSeries.timeDomain,
+        DomainPriority.GRIFF
+      );
+      return [...acc, completedSeries];
+    }, []);
+  };
+
+  registerCollection = ({ id, ...collection }: IncomingCollection) => {
```

Comment:
Can't `registerCollection` reuse `updateCollection`? Same with `registerSeries`?

Reply:
Yep, it looks like they can. They were originally different, but that's obsolete now.